share/p2p/shrexnd/server.go

```diff
@@ -0,0 +1,168 @@
+package shrexnd
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/minio/sha256-simd"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/eds"
+	"github.com/celestiaorg/celestia-node/share/getters"
+	"github.com/celestiaorg/celestia-node/share/ipld"
+	share_p2p_v1 "github.com/celestiaorg/celestia-node/share/p2p/shrexnd/v1/pb"
+	"github.com/celestiaorg/go-libp2p-messenger/serde"
+)
+
+const ndProtocolID = "/shrex/nd/0.0.1"
+
+var log = logging.Logger("shrex/nd")
+
+const (
+	writeTimeout = time.Second * 10
+	readTimeout  = time.Second * 10
+	serveTimeout = time.Second * 10
+)
+
+type Server struct {
+	getter       share.Getter
+	store        *eds.Store
+	host         host.Host
+	writeTimeout time.Duration
+	readTimeout  time.Duration
+	serveTimeout time.Duration
+}
+
+func StartServer(host host.Host, store *eds.Store, getter *getters.IPLDGetter) *Server {
```

Comment:
Actually, It does not start anything. it creates an instance of a server. So, I'd suggest to rename to a `NewServer`

Reply:
yes, + add godoc

Reply:
Also, I'd suggest extracting SetStreamHandler in `Start()`. I know, it will look redundant but we have Start/Stop pattern in all of our services, so can we follow it in shrexnd also?